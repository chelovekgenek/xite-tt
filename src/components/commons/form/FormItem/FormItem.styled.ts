import styled from "styled-components"

import { Form as AntForm } from "antd"

export const FormItem = styled(AntForm.Item)`
  &.ant-form-item {
    display: flex;
  }
  .ant-form-item-control-wrapper {
    width: 100%;
  }
`
