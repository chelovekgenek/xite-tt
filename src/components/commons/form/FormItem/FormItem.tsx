import React from "react"
import { isNil } from "lodash-es"

import { Grid } from "components/commons"

import * as Styled from "./FormItem.styled"

const { Row, Col } = Grid

interface IProps {
  label: string
  children: React.ReactNode
  touched?: any
  error?: any
}

export const FormItem: React.FC<IProps> = ({ children, label, touched, error }) => (
  <Row type="flex">
    <Col span={4}>{label}</Col>
    <Col span={20}>
      <Styled.FormItem
        {...(touched && !isNil(error)
          ? {
              validateStatus: "error",
              help: error,
            }
          : {})}
      >
        {children}
      </Styled.FormItem>
    </Col>
  </Row>
)
