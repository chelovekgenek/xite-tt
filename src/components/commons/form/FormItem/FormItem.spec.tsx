import React from "react"
import { shallow } from "enzyme"

import { FormItem } from "./FormItem"

describe("FormItem", () => {
  const render = (props = {}) => shallow(<FormItem label="asd" children={true} {...props} />)
  it("renders correctly", () => {
    const comp = render()
    expect(comp).toMatchSnapshot()
  })
  it("should render an error if form is touched and error is provided", () => {
    const comp = render({ touched: true, error: "error message" })
    expect(comp.find("Styled(FormItem)").props()).toEqual({
      validateStatus: "error",
      help: "error message",
      children: true,
    })
  })
})
