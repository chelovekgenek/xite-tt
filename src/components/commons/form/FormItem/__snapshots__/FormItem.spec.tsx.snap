// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem renders correctly 1`] = `
<Row
  gutter={0}
  type="flex"
>
  <Col
    span={4}
  >
    asd
  </Col>
  <Col
    span={20}
  >
    <Styled(FormItem)>
      <Component />
    </Styled(FormItem)>
  </Col>
</Row>
`;
